modules/targets/darwin/fonts.nix

```diff
@@ -10,20 +10,17 @@ let
     pathsToLink = "/share/fonts";
   };
   fonts = "${fontsEnv}/share/fonts";
+  installDir = "${homeDir}/Library/Fonts/HomeManager";
 in {
   # macOS won't recognize symlinked fonts
   config = mkIf pkgs.stdenv.hostPlatform.isDarwin {
-    home.activation.copyFonts = hm.dag.entryAfter [ "writeBoundary" ] ''
-      copyFonts() {
-        rm -rf ${homeDir}/Library/Fonts/HomeManager || :
-
-        local f
-        find -L "${fonts}" -type f -printf '%P\0' | while IFS= read -rd "" f; do
-          $DRY_RUN_CMD install $VERBOSE_ARG -Dm644 -T \
-            "${fonts}/$f" "${homeDir}/Library/Fonts/HomeManager/$f"
-        done
-      }
-      copyFonts
-    '';
+    home.file."Library/Fonts/.home-manager-fonts-version" = {
+      text = "${fontsEnv}";
+      onChange = ''
+        mkdir -p ${escapeShellArg installDir}
+        ${pkgs.rsync}/bin/rsync $VERBOSE_ARG -acL --chmod=u+w --delete \
+          ${escapeShellArgs ["${fonts}/" installDir]}
+      '';
```

Comment:
These probably still need to be protected by the `run` shell function.

https://nix-community.github.io/home-manager/release-notes.xhtml#sec-release-24.05-highlights

Reply:
Never mind, I see that `run` is also necessary to respect the `silence` flag. I've added it. :)